File: test/buf/c/src/buf.c
/*[[[cog
import cogging as c
c.tpl(cog,templateFile,c.a(prefix=configFile))
]]]*/

#include "../include/buf.h"

void* com_github_airutech_gernet_test_buf_readNext(bufferKernelParams *params, BOOL make_timeout);
bufferReadData com_github_airutech_gernet_test_buf_readNextWithMeta(bufferKernelParams *params, BOOL make_timeout);
int com_github_airutech_gernet_test_buf_readFinished(bufferKernelParams *params);
void* com_github_airutech_gernet_test_buf_writeNext(bufferKernelParams *params, BOOL make_timeout);
int com_github_airutech_gernet_test_buf_writeFinished(bufferKernelParams *params);
int com_github_airutech_gernet_test_buf_size(bufferKernelParams *params);
int com_github_airutech_gernet_test_buf_timeout(bufferKernelParams *params);
int com_github_airutech_gernet_test_buf_gridSize(bufferKernelParams *params);
int com_github_airutech_gernet_test_buf_uniqueId(bufferKernelParams *params);
int com_github_airutech_gernet_test_buf_addSelector(bufferKernelParams *params, void* selectorContainer);
void com_github_airutech_gernet_test_buf_onCreate(com_github_airutech_gernet_test_buf *that);
void com_github_airutech_gernet_test_buf_onDestroy(com_github_airutech_gernet_test_buf *that);

reader com_github_airutech_gernet_test_buf_getReader(com_github_airutech_gernet_test_buf *that, void* container, int grid_id){
  bufferKernelParams_create(params, that, grid_id, container,com_github_airutech_gernet_test_buf_)
  reader_create(res,params)
  return res;
}

writer com_github_airutech_gernet_test_buf_getWriter(com_github_airutech_gernet_test_buf *that, void* container, int grid_id){
  bufferKernelParams_create(params, that, grid_id, container,com_github_airutech_gernet_test_buf_)
  writer_create(res,params)
  return res;
}

void com_github_airutech_gernet_test_buf_initialize(com_github_airutech_gernet_test_buf *that){
  com_github_airutech_gernet_test_buf_onCreate(that);
}

void com_github_airutech_gernet_test_buf_deinitialize(struct com_github_airutech_gernet_test_buf *that){
  com_github_airutech_gernet_test_buf_onDestroy(that);
}
/*[[[end]]] (checksum: b5961eaa7a6623b8a788ec7e0f4c0e8c)*/

void com_github_airutech_gernet_test_buf_onCreate(com_github_airutech_gernet_test_buf *that){
  
  return;
}

void com_github_airutech_gernet_test_buf_onDestroy(com_github_airutech_gernet_test_buf *that){
  
  return;
}

void* com_github_airutech_gernet_test_buf_readNext(bufferKernelParams *params, BOOL make_timeout) {
  bufferReadData res = com_github_airutech_gernet_test_buf_readNextWithMeta(params, make_timeout);
  return res.data;
}

bufferReadData com_github_airutech_gernet_test_buf_readNextWithMeta(bufferKernelParams *params, BOOL make_timeout) {
  bufferReadData res;
  res.data = NULL;
  if(params == NULL){
    printf("ERROR: com_github_airutech_gernet_test_buf readNextWithMeta: params is NULL\n");
    return res;
  }
  com_github_airutech_gernet_test_buf *that = (com_github_airutech_gernet_test_buf*)params->target;
  if(that == NULL){
    printf("ERROR: com_github_airutech_gernet_test_buf readNextWithMeta: Some Input parameters are wrong\n");
    return res;
  }
  /*TODO:IMPLEMENTATION GOES HERE*/
  return res;
}

int com_github_airutech_gernet_test_buf_readFinished(bufferKernelParams *params) {
  if(params == NULL){
    printf("ERROR: com_github_airutech_gernet_test_buf readFinished: params is NULL\n");
    return -1;
  }
  com_github_airutech_gernet_test_buf *that = (com_github_airutech_gernet_test_buf*)params->target;
  if(that == NULL){
    printf("ERROR: com_github_airutech_gernet_test_buf readFinished: Some Input parameters are wrong\n");
    return -1;
  }
  /*TODO:IMPLEMENTATION GOES HERE*/
  return 0;
}

void* com_github_airutech_gernet_test_buf_writeNext(bufferKernelParams *params, BOOL make_timeout) {
  if(params == NULL){
    printf("ERROR: com_github_airutech_gernet_test_buf writeNext: params is NULL\n");
    return NULL;
  }
  com_github_airutech_gernet_test_buf *that = (com_github_airutech_gernet_test_buf*)params->target;
  void* res = NULL;
  if(that == NULL){
    printf("ERROR: com_github_airutech_gernet_test_buf writeNext: Some Input parameters are wrong\n");
    return res;
  }
  /*TODO:IMPLEMENTATION GOES HERE*/
  return res;
}

int com_github_airutech_gernet_test_buf_writeFinished(bufferKernelParams *params) {
  if(params == NULL){
    printf("ERROR: com_github_airutech_gernet_test_buf writeFinished: params is NULL\n");
    return -1;
  }
  com_github_airutech_gernet_test_buf *that = (com_github_airutech_gernet_test_buf*)params->target;
  if(that == NULL){
    printf("ERROR: buf writeFinished: Some Input parameters are wrong\n");
    return -1;
  };
  /*TODO:IMPLEMENTATION GOES HERE*/
  return 0;
}

int com_github_airutech_gernet_test_buf_size(bufferKernelParams *params){
  return 0;
}

int com_github_airutech_gernet_test_buf_timeout(bufferKernelParams *params){
  return 0;
}

int com_github_airutech_gernet_test_buf_gridSize(bufferKernelParams *params){
  return 0;
}

int com_github_airutech_gernet_test_buf_uniqueId(bufferKernelParams *params){
  return 0;
}

int com_github_airutech_gernet_test_buf_addSelector(bufferKernelParams *params, void* selectorContainer) {
  return 0;
}

void com_github_airutech_gernet_test_buf_onKernels(com_github_airutech_gernet_test_buf *that){

  return;
}

File: test/buf/java/src/main/java/com/github/airutech/gernet/test/buf/buf.java

package com.github.airutech.gernet.test.buf;

/*[[[cog
import cogging as c
c.tpl(cog,templateFile,c.a(prefix=configFile))
]]]*/

import com.github.airutech.cnets.types.*;
import com.github.airutech.cnets.readerWriter.*;
import com.github.airutech.cnets.queue.*;

public class buf implements readerWriterInterface{
  Object[] buffers;long timeout_milisec;String uniqueId;int readers_grid_size;int statsInterval;
  
  public buf(Object[] buffers,long timeout_milisec,String uniqueId,int readers_grid_size,int statsInterval){
    this.buffers = buffers;
    this.timeout_milisec = timeout_milisec;
    this.uniqueId = uniqueId;
    this.readers_grid_size = readers_grid_size;
    this.statsInterval = statsInterval;
    onCreate();
    initialize();
  }


public reader getReader(long grid_id){
  Object container = null;
  return new reader(new bufferKernelParams(this, grid_id, container));
}
public writer getWriter(long grid_id){
  Object container = null;
  return new writer(new bufferKernelParams(this, grid_id, container));
}
  private void initialize(){
    
    onKernels();
    
  }
  
/*[[[end]]] (checksum: 6b334e8a97c08853569e08a72f3ed27e)*/

  private void onCreate(){

  }

  private void onKernels(){

  }

  @Override
  public Object readNext(bufferKernelParams params, int waitThreshold) {
    return readNextWithMeta(params, waitThreshold).getData();
  }

  @Override
  public bufferReadData readNextWithMeta(bufferKernelParams params, int waitThreshold) {
    if(this != params.getTarget()){return null;}
    buf m = (buf)params.getTarget();
    bufferReadData res = new bufferReadData();
    if(m==null){
      System.err.print("ERROR: buf readNextWithMeta: Some Input parameters are wrong");
      return res;
    }
    /*TODO:IMPLEMENTATION GOES HERE*/
    return res;
  }

  @Override
  public int readFinished(bufferKernelParams params) {
    if(this != params.getTarget()){return -1;}
    buf m = (buf)params.getTarget();
    if(m==null){
      System.err.print("ERROR: buf readFinished: Some Input parameters are wrong");
      return -1;
    }
    /*TODO:IMPLEMENTATION GOES HERE*/
    return 0;
  }

  @Override
  public Object writeNext(bufferKernelParams params, int waitThreshold) {
    if(this != params.getTarget()){return null;}
    buf m = (buf)params.getTarget();
    Object res = null;
    if(m==null){
      System.err.print("ERROR: buf writeNext: Some Input parameters are wrong");
      return res;
    }
    /*TODO:IMPLEMENTATION GOES HERE*/
    return res;
  }

  @Override
  public int writeFinished(bufferKernelParams params) {
    if(this != params.getTarget()){return -1;}
    buf m = (buf)params.getTarget();
    if(m==null){
      System.err.print("ERROR: buf writeFinished: Some Input parameters are wrong");
      return -1;
    };
    /*TODO:IMPLEMENTATION GOES HERE*/
    return 0;
  }

  @Override
  public int size(bufferKernelParams params){
    return 0;
  }

  @Override
  public int timeout(bufferKernelParams params){
    return 0;
  }

  @Override
  public int gridSize(bufferKernelParams params){
    return 0;
  }

  @Override
  public String uniqueId(bufferKernelParams params){
    return "";
  }

  @Override
  public long getStatsInterval(bufferKernelParams buffer) {
    return 0;
  }

  @Override
  public int addSelector(bufferKernelParams params, Object selectorContainer) {
    return 0;
  }
}

